homeassistant/components/binary_sensor/xiaomi_aqara.py

```diff
@@ -311,6 +317,48 @@ def parse_data(self, data, raw_data):
             return False
 
 
+class XiaomiVibration(XiaomiBinarySensor):
+    """Representation of a Xiaomi Vibration Sensor."""
+
+    def __init__(self, device, name, data_key, xiaomi_hub):
+        """Initialize the XiaomiVibration."""
+        self._last_action = None
+        super().__init__(device, name, xiaomi_hub, data_key, None)
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {ATTR_LAST_ACTION: self._last_action}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
+    def parse_data(self, data, raw_data):
+        """Parse data sent by gateway."""
+        value = data.get(self._data_key)
+        if value is None:
+            return False
+
+        if value == 'vibrate':
+            movement_type = 'vibrate'
+        elif value == 'tilt':
+            movement_type = 'tilt'
+        elif value == "free_fall":
+            movement_type = 'free_fall'
+        else:
+            if value not in ('vibrate', 'tilt',  'free_fall'):
+                _LOGGER.warning("Unsupported movement_type detected: %s",
+                                value)
+                return False
+
+        self.hass.bus.fire('xiaomi_aqara.movement', {
+            'entity_id': self.entity_id,
+            'movement_type': movement_type
+        })
+        self._last_action = movement_type
+
+        return True
+
```

Comment:
Replace `parse_data` with something like
```
    def parse_data(self, data, raw_data):
        """Parse data sent by gateway."""
        value = data.get(self._data_key)
        if value not in ('vibrate', 'tilt',  'free_fall'):
            _LOGGER.warning("Unsupported movement_type detected: %s", value)
            return False

        self._hass.bus.fire('movement', {
            'entity_id': self.entity_id,
            'movement_type': value
        })
        self._last_action = value

        return True
```

Reply:
Is 'xiaomi_aqara.movement' used here on purpose?